src/net.js

```diff
@@ -1,97 +1,102 @@
 
 const Big = require('big.js');
+const July23 = new Date("2023-07-01")
 
-module.exports = function net(gross, nguoiPhuThuoc, region = 1) {
+module.exports = function net(gross, nguoiPhuThuoc = 0, region = 1, date = new Date("2023-07-01")) {
     const grossBig = new Big(gross);
+
     console.log("Gross: ", grossBig.toNumber());
     console.log("Dependents: ", nguoiPhuThuoc);
+    let regionMinWage;
+    let grossBasic;
+    // console.log('Date:', date.toISOString());
 
-   
-    var giamThueNguoiPhuThuoc = new Big(0);
-
-    if (nguoiPhuThuoc > 0) {
 
-        giamThueNguoiPhuThuoc = new Big(nguoiPhuThuoc *4400000);
-    }
+    if (date.getTime() >= July23.getTime()) {
 
-    console.log("So tien khau tru nguoi phu thuoc: ", giamThueNguoiPhuThuoc.toNumber());
+        grossBasic = new Big(1_800_000);
 
 
-    //Gross > 93_600_000? regionMinWage*20: gross*0.01
-    let regionMinWage;
+        if (region === 1) {
+            regionMinWage = new Big(4_680_000);
+        } else if (region === 2) {
+            regionMinWage = new Big(4_160_000);
+        } else if (region === 3) {
+            regionMinWage = new Big(3_640_000);
+        } else if (region === 4) {
+            regionMinWage = new Big(3_250_000);
+        } else {
+            console.log("Invalid region entered. Please enter again! (1, 2, 3, 4) ");
+        }
 
-    if (region === 1) {
-        regionMinWage = 4680000;
-    } else if (region === 2) {
-        regionMinWage = 4160000;
-    } else if (region === 3) {
-        regionMinWage = 3640000;
-    } else if (region === 4) {
-        regionMinWage = 3250000;
     } else {
-        console.log("Invalid region entered. Please enter again! (1, 2, 3, 4) ")
-    };
+        grossBasic = new Big(1_490_000);
+
+        if (region === 1) {
+            regionMinWage = new Big(4_420_000);
+        } else if (region === 2) {
+            regionMinWage = new Big(3_920_000);
+        } else if (region === 3) {
+            regionMinWage = new Big(3_430_000);
+        } else if (region === 4) {
+            regionMinWage = new Big(3_070_000);
+        } else {
+            console.log("Invalid region entered. Please enter again! (1, 2, 3, 4) ");
+        }
+    }
+
+    const giamThueNguoiPhuThuoc = new Big(nguoiPhuThuoc * 4_400_000);
+
 
     // Tính toán các khoản bảo hiểm xã hội
-    const maxGrossForInsurance = new Big(36000000);
-    const maxGrossAccidental = new Big(regionMinWage * 20);
+    const maxGrossForInsurance = new Big(grossBasic.times(new Big(20)));
+    const maxGrossAccidental = new Big(regionMinWage.times(new Big(20)));
 
-    const BHXH = Math.min(maxGrossForInsurance.times(0.08), grossBig.times(0.08));
-    console.log("BHXH: ", BHXH);
+    const BHXH = maxGrossForInsurance.gt(grossBig) ? Big(grossBig.times(0.08)) : Big(maxGrossForInsurance.times(0.08));
```

Comment:
Pls change variable name to English. It is social insurance. Pls look for the rest of insurance